torch/csrc/jit/operator_upgraders/utils.cpp

```diff
@@ -0,0 +1,53 @@
+#include <torch/csrc/jit/operator_upgraders/utils.h>
+
+#include <c10/util/Optional.h>
+#include <torch/csrc/jit/operator_upgraders/version_map.h>
+#include <iostream>
+#include <regex>
+#include <string>
+#include <vector>
+
+namespace torch {
+namespace jit {
+
+c10::optional<UpgraderEntry> findUpgrader(
+    const std::vector<UpgraderEntry>& upgraders_for_schema,
+    size_t current_version) {
+  // we want to find the entry which satisfies following two conditions:
+  //    1. the version entry must be greater than current_version
+  //    2. Among the version entries, we need to see if the current version
+  //       is in the upgrader name range
+  auto pos = std::find_if(
+      upgraders_for_schema.begin(),
+      upgraders_for_schema.end(),
+      [current_version](const UpgraderEntry& entry) {
+        return entry.bumped_at_version > current_version;
+      });
```

Comment:
This assumes upgrader entries are sorted by `bumped_at_version`. Is this guaranteed?

Reply:
Yes. We can explitcitly mention it should be sorted. Since the version map is used frequently, I feel like it would be expensive to keep sorting it. But it is probably better if the developers maintain the sorted order

Reply:
If developers are expected to sort the order, it will be good to add a test to prevent adding unsorted entry.

Reply:
I think a unittest makes more sense. Because we add values to this map only for testing purposes.

Reply:
Now we have unittest, and also a sorting mechanism